compiler/resolution/functionResolution.cpp

```diff
@@ -4198,7 +4219,7 @@ static Expr* dropUnnecessaryCast(CallExpr* call) {
     INT_FATAL("dropUnnecessaryCasts called on non _cast call");
 
   if (SymExpr* sym = toSymExpr(call->get(2))) {
-    if (VarSymbol* var = toVarSymbol(sym->var)) {
+    if (LcnSymbol* var = toLcnSymbol(sym->var)) {
```

Comment:
Assuming this is a bug fix? If it's LcnSymbol, then it could be ArgSymbol or VarSymbol?
Could this change be committed separately to master?